fdbclient/command_runner.go

```diff
@@ -0,0 +1,64 @@
+/*
+ * command_runner.go
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2022 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package fdbclient
+
+import (
+	"context"
+	"github.com/go-logr/logr"
+	"os/exec"
+)
+
+// commandRunner is an interface to run commands.
+type commandRunner interface {
```

Comment:
This setup allows us to "run" whole commands and check if they behave like expected based on the mocked output. For everything that talks directly with FDB. without using a binary we have to implement a similar approach.